src/entities/college.entity.ts

```diff
@@ -0,0 +1,45 @@
+import { 
+  Entity, 
+  PrimaryGeneratedColumn, 
+  Column, 
+  CreateDateColumn, 
+  UpdateDateColumn, 
+  DeleteDateColumn, } from "typeorm";
+
+import { v4 as uuidv4 } from 'uuid';
+
+@Entity()
+export class College {
+  @PrimaryGeneratedColumn("uuid")
+  id!: string;
+
+  @Column({ type: "varchar", length: 50, nullable: false })
+  name!: string;
+
+  @Column({ type: "varchar", nullable: true })
+  uniform?: string;
+
+  @Column({ type: "date", nullable: false })
+  yearEstablished!: Date;
+
+  @Column({ type: "text", nullable: true })
+  description?: string;
+
+  @CreateDateColumn()
+  created_at!: Date;
+
+  @UpdateDateColumn()
+  updated_at!: Date;
+
+  @DeleteDateColumn()
+  deleted_at!: Date;
+}
+
+
```

Comment:
camel case is the normal convention for js/ts (na python and sql people prefer snake case). so they should be deletedAt, createdAt etc. then have a name parameter to show whats in the db is deleted_at